web/api/common/spotifyclient.go

```diff
@@ -0,0 +1,90 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type spotifyClient struct {
+	sync.WaitGroup
```

Comment:
Make this `wg sync.WaitGroup`, it's properties shouldn't be on the client object directly.